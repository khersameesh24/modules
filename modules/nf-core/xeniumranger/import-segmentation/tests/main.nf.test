nextflow_process {

    name "Test Process XENIUMRANGER_IMPORT_SEGMENTATION"
    script "../main.nf"
    process "XENIUMRANGER_IMPORT_SEGMENTATION"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "xeniumranger"
    tag "xeniumranger/import-segmentation"
    tag "untar"

    setup {
            run("UNTAR") {
                script "modules/nf-core/untar/main.nf"
                process {
                    """
                    input[0] = [[], file('https://raw.githubusercontent.com/nf-core/test-datasets/spatialxe/Xenium_Prime_Mouse_Ileum_tiny_outs.tar.gz', checkIfExists: true)]
                    """
                }
            }
    }

    test("xeniumranger import-segmentation nuclei npy") {
        when {
            process {
                """
                input[0] = UNTAR.out.untar.map{ meta, files -> [ [ id:'test_xeniumranger_import_segmentation' ], file(files)]}
                input[1] = []
                input[2] = UNTAR.out.untar.map { it[1] } + "/segmentations/nuclei.npy"
                input[3] = []
                input[4] = []
                input[5] = []
                input[6] = "pixels"
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.bundle.get(0).get(1).findAll { file(it).name !in [
                        'analysis_summary.html',
                        'metrics_summary.csv',
                        'cell_boundaries.csv.gz',
                        'cell_boundaries.parquet',
                        'nucleus_boundaries.csv.gz',
                        'nucleus_boundaries.parquet',
                        'cells.csv.gz',
                        'cells.parquet',
                        'cells.zarr.zip',
                        'transcripts.parquet',
                        'transcripts.zarr.zip',
                        'clusters.csv',
                        'differential_expression.csv',
                        'components.csv',
                        'projection.csv',
                        'variance.csv',
                        'analysis.zarr.zip',
                        'experiment.xenium',
                        'cell_feature_matrix.zarr.zip'
                        ]}
                    ).match()
                },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'analysis_summary.html' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'metrics_summary.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cell_boundaries.csv.gz' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cell_boundaries.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'nucleus_boundaries.csv.gz' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'nucleus_boundaries.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cells.csv.gz' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cells.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cells.zarr.zip' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'transcripts.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'clusters.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'differential_expression.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'components.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'projection.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'variance.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'analysis.zarr.zip' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'experiment.xenium' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cell_feature_matrix.zarr.zip' }).exists() },
            )
        }
    }


    test("xeniumranger import-segmentation nuclei tif") {
        when {
            process {
                """
                input[0] = UNTAR.out.untar.map{ meta, files -> [ [ id:'test_xeniumranger_import_segmentation' ], file(files)]}
                input[1] = []
                input[2] = UNTAR.out.untar.map { it[1] } + "/segmentations/nuclei.tif"
                input[3] = []
                input[4] = []
                input[5] = []
                input[6] = "pixels"
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.bundle.get(0).get(1).findAll { file(it).name !in [
                        'analysis_summary.html',
                        'metrics_summary.csv',
                        'cell_boundaries.csv.gz',
                        'cell_boundaries.parquet',
                        'nucleus_boundaries.csv.gz',
                        'nucleus_boundaries.parquet',
                        'cells.csv.gz',
                        'cells.parquet',
                        'cells.zarr.zip',
                        'transcripts.parquet',
                        'transcripts.zarr.zip',
                        'clusters.csv',
                        'differential_expression.csv',
                        'components.csv',
                        'projection.csv',
                        'variance.csv',
                        'analysis.zarr.zip',
                        'experiment.xenium',
                        'cell_feature_matrix.zarr.zip'
                        ]}
                    ).match()
                },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'analysis_summary.html' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'metrics_summary.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cell_boundaries.csv.gz' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cell_boundaries.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'nucleus_boundaries.csv.gz' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'nucleus_boundaries.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cells.csv.gz' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cells.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cells.zarr.zip' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'transcripts.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'clusters.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'differential_expression.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'components.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'projection.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'variance.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'analysis.zarr.zip' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'experiment.xenium' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cell_feature_matrix.zarr.zip' }).exists() },
            )
        }
    }

    test("xeniumranger import-segmentation segmentation csv") {
        when {
            process {
                """
                input[0] = UNTAR.out.untar.map{ meta, files -> [ [ id:'test_xeniumranger_import_segmentation' ], file(files)]}
                input[1] = UNTAR.out.untar.map { it[1] } + "/segmentations/imagealignment.csv"
                input[2] = []
                input[3] = []
                input[4] = UNTAR.out.untar.map { it[1] } + "/segmentations/segmentation.csv"
                input[5] = UNTAR.out.untar.map { it[1] } + "/segmentations/segmentation_polygons.json"
                input[6] = "microns"
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.bundle.get(0).get(1).findAll { file(it).name !in [
                        'analysis_summary.html',
                        'metrics_summary.csv',
                        'cell_boundaries.csv.gz',
                        'cell_boundaries.parquet',
                        'nucleus_boundaries.csv.gz',
                        'nucleus_boundaries.parquet',
                        'cells.csv.gz',
                        'cells.parquet',
                        'cells.zarr.zip',
                        'transcripts.parquet',
                        'transcripts.zarr.zip',
                        'clusters.csv',
                        'differential_expression.csv',
                        'components.csv',
                        'projection.csv',
                        'variance.csv',
                        'analysis.zarr.zip',
                        'experiment.xenium',
                        'cell_feature_matrix.zarr.zip'
                        ]}
                    ).match()
                },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'analysis_summary.html' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'metrics_summary.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cell_boundaries.csv.gz' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cell_boundaries.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'nucleus_boundaries.csv.gz' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'nucleus_boundaries.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cells.csv.gz' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cells.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cells.zarr.zip' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'transcripts.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'clusters.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'differential_expression.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'components.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'projection.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'variance.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'analysis.zarr.zip' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'experiment.xenium' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cell_feature_matrix.zarr.zip' }).exists() },
            )
        }
    }

    test("xeniumranger import-segmentation") {
        when {
            process {
                """
                input[0] = UNTAR.out.untar.map{ meta, files -> [ [ id:'test_xeniumranger_import_segmentation' ], file(files)]}
                input[1] = 0
                input[2] = []
                input[3] = UNTAR.out.untar.map { it[1] } + "/cells.zarr.zip"
                input[4] = []
                input[5] = []
                input[6] = ""
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.bundle.get(0).get(1).findAll { file(it).name !in [
                        'analysis_summary.html',
                        'metrics_summary.csv',
                        'cell_boundaries.csv.gz',
                        'cell_boundaries.parquet',
                        'nucleus_boundaries.csv.gz',
                        'nucleus_boundaries.parquet',
                        'cells.csv.gz',
                        'cells.parquet',
                        'cells.zarr.zip',
                        'transcripts.parquet',
                        'transcripts.zarr.zip',
                        'clusters.csv',
                        'differential_expression.csv',
                        'components.csv',
                        'projection.csv',
                        'variance.csv',
                        'analysis.zarr.zip',
                        'experiment.xenium',
                        'cell_feature_matrix.zarr.zip'
                        ]}
                    ).match()
                },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'analysis_summary.html' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'metrics_summary.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cell_boundaries.csv.gz' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cell_boundaries.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'nucleus_boundaries.csv.gz' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'nucleus_boundaries.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cells.csv.gz' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cells.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cells.zarr.zip' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'transcripts.parquet' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'clusters.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'differential_expression.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'components.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'projection.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'variance.csv' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'analysis.zarr.zip' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'experiment.xenium' }).exists() },
                { assert file(process.out.bundle.get(0).get(1).find { file(it).name == 'cell_feature_matrix.zarr.zip' }).exists() },
            )
        }
    }

    test("xeniumranger import-segmentation stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = UNTAR.out.untar.map{ meta, files -> [ [ id:'test_xeniumranger_import_segmentation' ], file(files)]}
                input[1] = 0
                input[2] = []
                input[3] = UNTAR.out.untar.map { it[1] } + "/cells.zarr.zip"
                input[4] = []
                input[5] = []
                input[6] = ""
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}