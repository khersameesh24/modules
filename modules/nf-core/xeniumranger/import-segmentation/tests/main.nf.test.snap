{
    "xeniumranger import-segmentation": {
        "content": [
            [
                "versions.yml:md5,d76e870d71abf94ed9ae972a08b83f63"
            ],
            [
                "dispersion.csv:md5,e8b1abb880ece8fb730ce34a15f958b4",
                "features_selected.csv:md5,c5e32d69f001f938ed316d2108a21e00",
                "cell_feature_matrix.h5:md5,96cb400f1b1dd6f8796daea0ad5c74e6",
                "barcodes.tsv.gz:md5,04ea06796d6b28517c288904ca043582",
                "features.tsv.gz:md5,7862242129681900a9cc4086dc83b62e",
                "matrix.mtx.gz:md5,489f86fbd8d65d6b973bb9cc7c5a76f1",
                "gene_panel.json:md5,8890dd5fd90706e751554ac3fdfdedde",
                "morphology.ome.tif:md5,6b65fff28a38a001b8f25061737fbf9b",
                "morphology_focus_0000.ome.tif:md5,90e796ad634d14e62cf2ebcadf2eaf98"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-30T00:13:13.575888"
    },
    "xeniumranger import-segmentation nuclei npy": {
        "content": [
            [
                "versions.yml:md5,063a1e5a49d24bdb9c7205d9a9232fdb"
            ],
            [
                "/",
                "w",
                "o",
                "r",
                "k",
                "s",
                "p",
                "a",
                "c",
                "e",
                "/",
                "m",
                "o",
                "d",
                "u",
                "l",
                "e",
                "s",
                "/",
                ".",
                "n",
                "f",
                "-",
                "t",
                "e",
                "s",
                "t",
                "/",
                "t",
                "e",
                "s",
                "t",
                "s",
                "/",
                "d",
                "4",
                "4",
                "d",
                "f",
                "0",
                "4",
                "e",
                "1",
                "5",
                "6",
                "5",
                "8",
                "d",
                "9",
                "2",
                "a",
                "f",
                "5",
                "1",
                "0",
                "c",
                "0",
                "1",
                "4",
                "e",
                "d",
                "7",
                "f",
                "b",
                "3",
                "c",
                "/",
                "w",
                "o",
                "r",
                "k",
                "/",
                "3",
                "c",
                "/",
                "f",
                "7",
                "1",
                "0",
                "9",
                "5",
                "f",
                "9",
                "1",
                "6",
                "2",
                "e",
                "a",
                "c",
                "9",
                "8",
                "1",
                "7",
                "2",
                "4",
                "f",
                "9",
                "1",
                "6",
                "8",
                "5",
                "b",
                "1",
                "c",
                "9",
                "/",
                "t",
                "e",
                "s",
                "t",
                "_",
                "x",
                "e",
                "n",
                "i",
                "u",
                "m",
                "r",
                "a",
                "n",
                "g",
                "e",
                "r",
                "_",
                "i",
                "m",
                "p",
                "o",
                "r",
                "t",
                "_",
                "s",
                "e",
                "g",
                "m",
                "e",
                "n",
                "t",
                "a",
                "t",
                "i",
                "o",
                "n",
                "/",
                "o",
                "u",
                "t",
                "s"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-02T00:41:51.375690482"
    },
    "xeniumranger import-segmentation segmentation csv": {
        "content": [
            [
                "versions.yml:md5,d76e870d71abf94ed9ae972a08b83f63"
            ],
            [
                "dispersion.csv:md5,e8b1abb880ece8fb730ce34a15f958b4",
                "features_selected.csv:md5,c5e32d69f001f938ed316d2108a21e00",
                "cell_feature_matrix.h5:md5,5d74ea595561e0300b6c3e5ec8d06fff",
                "barcodes.tsv.gz:md5,97496a9b448d9380cff0575b8e7a6f57",
                "features.tsv.gz:md5,7862242129681900a9cc4086dc83b62e",
                "matrix.mtx.gz:md5,f93ed82a2a74c154392fc6237642f1d2",
                "gene_panel.json:md5,8890dd5fd90706e751554ac3fdfdedde",
                "morphology.ome.tif:md5,6b65fff28a38a001b8f25061737fbf9b",
                "morphology_focus_0000.ome.tif:md5,90e796ad634d14e62cf2ebcadf2eaf98"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-29T23:22:58.158857"
    },
    "xeniumranger import-segmentation stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "bundle": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-02T00:30:31.371781976"
    },
    "xeniumranger import-segmentation nuclei tif": {
        "content": [
            [
                "versions.yml:md5,d76e870d71abf94ed9ae972a08b83f63"
            ],
            [
                "dispersion.csv:md5,e8b1abb880ece8fb730ce34a15f958b4",
                "features_selected.csv:md5,c5e32d69f001f938ed316d2108a21e00",
                "cell_feature_matrix.h5:md5,96cb400f1b1dd6f8796daea0ad5c74e6",
                "barcodes.tsv.gz:md5,04ea06796d6b28517c288904ca043582",
                "features.tsv.gz:md5,7862242129681900a9cc4086dc83b62e",
                "matrix.mtx.gz:md5,489f86fbd8d65d6b973bb9cc7c5a76f1",
                "gene_panel.json:md5,8890dd5fd90706e751554ac3fdfdedde",
                "morphology.ome.tif:md5,6b65fff28a38a001b8f25061737fbf9b",
                "morphology_focus_0000.ome.tif:md5,90e796ad634d14e62cf2ebcadf2eaf98"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-29T23:11:37.18721"
    }
}